src/plots/declarative.py

```diff
@@ -59,30 +106,61 @@ def create_figure(self):
             'yticklabels': self._set_yticklabels,
             'invert_xaxis': self._invert_xaxis,
             'invert_yaxis': self._invert_yaxis,
-            'yscale': self._set_yscale
+            'yscale': self._set_yscale,
+            'map_features': self._add_map_features
         }
 
-        try:
-            axes = self.axes.flatten()
-        except AttributeError:
-            axes = [self.axes]
+        if feature in feature_dict:
+            feature_dict[feature](ax, vars(plot_obj)[feature])
 
-        #Loop through axis:
-        for i, ax in enumerate(axes):
-            # Grab correctly indexed plot object
-            plot_obj = self.plot_list[i]
+    def _map_scatter(self, plotobj, ax):
 
-            # Loop through plot layers
-            for layer in plot_obj.plot_layers:
-                plot_dict[layer.plottype](layer, ax)
+        if plotobj.data is None:
+            skipvars = ['plottype', 'longitude', 'latitude',
+                        'markersize']
+            inputs = self._get_inputs_dict(skipvars, plotobj)
 
-            # loop through all keys in an object and then call approriate
-            # method to plot the feature on the axis
-            for feat in vars(plot_obj).keys():
-                if feat in feature_dict:
-                    feature_dict[feat](ax, vars(plot_obj)[feat])
+            cs = ax.scatter(plotobj.longitude, plotobj.latitude,
+                            s=plotobj.markersize, **inputs,
+                            transform=self.projection.projection)
+        else:
+            skipvars = ['plottype', 'longitude', 'latitude',
+                        'data', 'markersize']
+            inputs = self._get_inputs_dict(skipvars, plotobj)
+            cs = ax.scatter(plotobj.longitude, plotobj.latitude,
+                            c=plotobj.data, s=plotobj.markersize,
+                            **inputs, transform=self.projection.projection)
+#         if plotobj.colorbar:
```

Comment:
comments?

Reply:
I guess colorbar isn't working yet, did I miss that?

Reply:
Yes, still working out the kinks of colorbar. It gets tricky, I will continue to work on this tomorrow.